src/Servers/Kestrel/Core/src/Internal/CertificatePathWatcher.cs

```diff
@@ -29,9 +29,17 @@ internal sealed partial class CertificatePathWatcher : IDisposable
 
     public CertificatePathWatcher(IHostEnvironment hostEnvironment, ILogger<CertificatePathWatcher> logger)
         : this(
-              hostEnvironment.ContentRootPath,
-              logger,
-              dir => Directory.Exists(dir) ? new PhysicalFileProvider(dir, ExclusionFilters.None) : null)
+            hostEnvironment.ContentRootPath,
+            logger,
+            dir => Directory.Exists(dir)
+                ? new PhysicalFileProvider(dir, ExclusionFilters.None)
+                {
+                    // Force polling because it monitors both symlinks and their targets,
+                    // whereas the non-polling watcher only monitors the symlinks themselves
+                    UseActivePolling = true,
```

Comment:
Do we know the performance impact of *always* doing this?

Reply:
Polling isn't on by default (for the IHostingEnvironment.ContentRootFileProvider), I'm not sure it's about the number of apps, it might be the number of files being polled in the common case. Is this just polling the path specified by the cert in the 90% scenario?

Reply:
My point is that this change always turns on file watching no matter what. That’s not something we’ve tested before. I just wanted to understand the scope of the polling.

Reply:
If ReloadOnChange is true, then we will poll files in your (non-code) configuration for the default certificate and for each endpoint's certificate and SNI certificates.  Obviously, this only applies to file-based certs (vs from the store).  If the same certificate file appears in more than one of those locations, it will only be polled once.  (There's a corner case where two different symlinks point to the same file, in which case it would be polled twice.)